aten/src/ATen/DynamicLibrary.cpp

```diff
@@ -46,11 +46,31 @@ DynamicLibrary::~DynamicLibrary() {
 
 DynamicLibrary::DynamicLibrary(const char* name) {
   // NOLINTNEXTLINE(hicpp-signed-bitwise)
-  HMODULE theModule = LoadLibraryA(name);
+  HMODULE theModule;
+  bool reload = true;
+  if (GetProcAddress(GetModuleHandle("KERNEL32.DLL"), "AddDllDirectory") != NULL) {
```

Comment:
Please add a comment explaining why this is needed
```suggestion
  // Check if LOAD_LIBRARY_SEARCH_DEFAULT_DIRS is supported
  if (GetProcAddress(GetModuleHandle("KERNEL32.DLL"), "AddDllDirectory") != NULL) {
```

Reply:
Sounds reasonable.